### PR TITLE
feat(status line themes): Add predefined status line themes - powerline, pill shape, etc.

### PR DESCRIPTION
**Pull request by VeejayPL**
Added a few status line themes to choose from, apart from the default half pill design. Hopefully, this meets users needs when it comes to customizing the status line appearance.
Relates to PR: #27 

`set -g @catppuccin_pill_theme_enabled on`
pill

`set -g @catppuccin_powerline_theme_enabled on`
`set -g @catppuccin_l_left_separator ""`
`set -g @catppuccin_l_right_separator ""`
`set -g @catppuccin_r_left_separator ""`
`set -g @catppuccin_r_right_separator ""`
powerline

`set -g @catppuccin_powerline_icons_theme_enabled on`
`set -g @catppuccin_l_left_separator ""`
`set -g @catppuccin_l_right_separator ""`
`set -g @catppuccin_r_left_separator ""`
`set -g @catppuccin_r_right_separator ""`
powerline-icons

`set -g @catppuccin_no_patched_fonts_theme_enabled on`
no-patched-fonts

There can be further customization by specifying the separators for left and right status with the following:

set -g @catppuccin_l_left_separator "<your_separator>"
set -g @catppuccin_l_right_separator "<your_separator>"
set -g @catppuccin_r_left_separator "<your_separator>"
set -g @catppuccin_r_right_separator "<your_separator>"

Of course there will be a different behaviour when specifying them for either a powerline or pill theme.

Updated README accordingly. 
Since the file grows, I've added table of contents for better navigation. Also, fixed some typos.

### CODE REVIEW

File: src/default.conf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+local show_directory
+readonly show_directory="#[fg=$thm_pink,bg=$thm_bg,nobold,nounderscore,noitalics]$r_left_separator#[fg=$thm_bg,bg=$thm_pink,nobold,nounderscore,noitalics]  #[fg=$thm_fg,bg=$thm_gray] #{b:pane_current_path} "
```
Comment (adaviloper) [inline src/default.conf:2]: The directory icon here seems to have an extra space compared to the other icons in this block.

Comment (rogeruiz) [reply src/default.conf]: Good catch on this. Looks like it was also caught in #49 as well.

**Comment by VeejayPL** [comment]
Could you please let me know if I should work on this to match the mainstream @rogeruiz?  
The icons for the separator can also be customized here but I don’t think they will depreciate any time soon.
#49 is already included here.

**Review by nnhutan — APPROVED** [review]

**Comment by rogeruiz** [comment]
I'll take a look at this today to get it merged in. Thanks for the ping @nnhutan and sorry @VeejayPL for not getting this in sooner.

**Review by the-spanish-guy — APPROVED** [review]

**Comment by rogeruiz** [comment]
Great work on this @VeejayPL 🚀

**Comment by VeejayPL** [comment]
Thank you @rogeruiz!

**Pull request merged**